tests/by-util/test_cat.rs

```diff
@@ -347,7 +347,13 @@ fn test_squeeze_blank_before_numbering() {
 #[cfg(unix)]
 fn test_dev_random() {
     let mut buf = [0; 2048];
-    let mut proc = new_ucmd!().args(&["/dev/random"]).run_no_wait();
+    #[cfg(target_os = "linux")]
+    fn rand_gen() -> &'static str { "/dev/urandom"}
+
+    #[cfg(not(target_os = "linux"))]
+    fn rand_gen() -> &'static str { "/dev/random"}
+
+    let mut proc = new_ucmd!().args(&[rand_gen()]).run_no_wait();
```

Comment:
I'm not sure about what the best practise here is, however in this project I've mostly seen constant strings used like this:
```rust
#[cfg(target_os = "linux")]
const DEV_RANDOM: &str = "/dev/urandom";
#[cfg(not(target_os = "linux"))]
const DEV_RANDOM: &str = "/dev/random";

let mut proc = new_ucmd!().arg(DEV_RANDOM).run_no_wait();

```

Reply:
I refactored to the const syntax.